Quartz.Spi.MongoDbJobStore.Tests/BaseStoreTests.cs

```diff
@@ -12,17 +12,19 @@ public abstract class BaseStoreTests
 
         protected IScheduler CreateScheduler(string instanceName = "QUARTZ_TEST")
         {
-            var properties = new NameValueCollection();
-            properties[StdSchedulerFactory.PropertySchedulerInstanceName] = instanceName;
-            properties[StdSchedulerFactory.PropertySchedulerInstanceId] = $"{Environment.MachineName}-{Guid.NewGuid()}";
-            properties[StdSchedulerFactory.PropertyJobStoreType] = typeof (MongoDbJobStore).AssemblyQualifiedName;
-            properties[
-                $"{StdSchedulerFactory.PropertyJobStorePrefix}.{StdSchedulerFactory.PropertyDataSourceConnectionString}"
-                ] = "mongodb://localhost/quartz";
-            properties[$"{StdSchedulerFactory.PropertyJobStorePrefix}.collectionPrefix"] = "prefix";
+            var properties = new NameValueCollection
+            {
+                ["quartz.serializer.type"] = "binary",
+                [StdSchedulerFactory.PropertySchedulerInstanceName] = instanceName,
+                [StdSchedulerFactory.PropertySchedulerInstanceId] = $"{Environment.MachineName}-{Guid.NewGuid()}",
+                [StdSchedulerFactory.PropertyJobStoreType] = typeof(MongoDbJobStore).AssemblyQualifiedName,
+                [$"{StdSchedulerFactory.PropertyJobStorePrefix}.{StdSchedulerFactory.PropertyDataSourceConnectionString}"]
+                    = "mongodb://localhost/quartz",
+                [$"{StdSchedulerFactory.PropertyJobStorePrefix}.collectionPrefix"] = "prefix"
+            };
 
             var scheduler = new StdSchedulerFactory(properties);
-            return scheduler.GetScheduler();
+            return scheduler.GetScheduler().Result;
```

Comment:
In general, .Wait() means block until we're done, and if the task returns a result type, discard the result. On the other hand .Result means wait until the result is ready and then we get the value. It's how we "unwrap" a Task<T> into a T. If the task was cancelled, or the execution of the task threw an exception, then attempting to .Result will throw an AggregateException where the InnerException contains the actual exception from the task.

My main goal with the .Wait() and .Result stuff is to make sure that all Tasks are still handled in sequential order and we don't ever orphan anything by returning from a function before underlying tasks have been completed (since I believe that's undefined behavior).